### PR TITLE
feat: track prepare next epoch time

### PR DESCRIPTION
**Pull request by twoeths**
**Motivation**

Track prepare next epoch time at lodestar side, the log shows that a lot of time it take >4s on holesky

**Description**

- This is to track how much time does it take at lodestar side to run epoch transition + `hashTreeRoot()` 
- There are also time to call execution engine, but I only want to focus on lodestar side  in order to improve performance

### CODE REVIEW

File: packages/beacon-node/src/metrics/metrics/lodestar.ts [hunk 1]
```diff
@@ -1298,6 +1298,11 @@ export function createLodestarMetrics(
         name: "lodestar_precompute_next_epoch_transition_waste_total",
         help: "Total number of precomputing next epoch transition wasted",
       }),
+      duration: register.histogram({
+        name: "lodestar_precompute_next_epoch_transition_duration_seconds",
+        help: "Duration of precomputeNextEpochTransition at lodestar side",
```
Comment (nflaig) [inline packages/beacon-node/src/metrics/metrics/lodestar.ts]: The "at lodestar side" might be confusing, are we precomputing the next epoch transition somewhere else (not in lodestar)?

Looking at the code the two major tasks we do are
- precompute next epoch transition
- and prepare execution payload

Might be missing something

Comment (wemeetagain) [reply packages/beacon-node/src/metrics/metrics/lodestar.ts]: There's also the prepareExecutionPayload call that's not included in this timer.

Comment (nflaig) [reply packages/beacon-node/src/metrics/metrics/lodestar.ts]: But isn't that a different task or maybe I misunderstand what is meant by `precomputeEpochTransition`, my assumption is this includes epoch transition + hashing the state.

From my understanding the "at lodestar side" is redundant and can be removed as the task `precomputeNextEpochTransition` is only executed by Lodestar.

`prepareExecutionPayload` is part of preparing for next slot but not part of precomputing the epoch transition.

Comment (wemeetagain) [inline packages/beacon-node/src/metrics/metrics/lodestar.ts]:
```suggestion
        help: "Duration of precomputeNextEpochTransition, including epoch transition and hashTreeRoot",
```

**Review by wemeetagain — DISMISSED** [review]

File: packages/beacon-node/src/chain/prepareNextSlot.ts [hunk 2]
```diff
@@ -97,6 +97,9 @@ export class PrepareNextSlotScheduler {
         headRoot,
         isEpochTransition,
       });
+      const lodestarPrecomputeEpochTransitionTimer = isEpochTransition
```
Comment (nflaig) [inline packages/beacon-node/src/chain/prepareNextSlot.ts]:
```suggestion
      const precomputeEpochTransitionTimer = isEpochTransition
```

File: packages/beacon-node/src/chain/prepareNextSlot.ts [hunk 3]
```diff
@@ -133,6 +136,8 @@ export class PrepareNextSlotScheduler {
           prepareSlot,
           previousHits,
         });
+
+        lodestarPrecomputeEpochTransitionTimer?.();
```
Comment (nflaig) [inline packages/beacon-node/src/chain/prepareNextSlot.ts]:
```suggestion
        precomputeEpochTransitionTimer?.();
```

**Review by nflaig — APPROVED** [review]

**Review by wemeetagain — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.14.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.14.0) :tada:

**Pull request merged**